src/target/riscv/riscv-013.c

```diff
@@ -2051,6 +2077,9 @@ static int examine(struct target *target)
 		else
 			LOG_TARGET_INFO(target, "Core %d could not be made part of halt group %d.",
 					info->index, target->smp);
+		if (r->external_trigger)
```

Comment:
1. it's not quite clear why this code is under `target->smp` condition. We can control an external triggger without creating an SMP group.
2. @en-sc I remember you wanted to implement an interface to debug module. Could you please assess if we can implement the functionality in question without this interface?